### PR TITLE
docs: Add `clients` subdirective to `tls` directive docs

### PR DESCRIPTION
**Pull request by danlsgiga**
This PR update the documentation for the Caddyfile `tls` directive for `v2.1`.

This basically adds the `clients` sub-directive which is being added in `v2.1` via [this PR](https://github.com/caddyserver/caddy/pull/3335)

The `clients` subdirective enables TLS Client Authentication using the Caddyfile.

### CODE REVIEW

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (francislavoie) [inline src/docs/markdown/caddyfile/directives/tls.md]: Spaces -> tabs

```suggestion
tls {
	clients {
		mode                    require_and_verify
		trusted_ca_certs_file   cacerts.crt
	}
}
```

**Comment by francislavoie** [comment]
Thanks, this is great! We'll wait until the other PR is merged to merge this, naturally.

File: src/docs/markdown/caddyfile/directives/reverse_proxy.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/reverse_proxy.md:131]:
```suggestion
	tls_server_name <sni>
```

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:33]: The syntax has changed a little bit recently: https://github.com/caddyserver/caddy/pull/3335/files#diff-3235e86b75f6440c822f6d7d8f4beb61R65-R71

Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:33]:
```suggestion
	client_auth {
		mode                   [request|require|verify_if_given|require_and_verify]
		trusted_ca_cert        <base64_der>
		trusted_ca_cert_file   <filename>
		trusted_leaf_cert      <base64_der>
		trusted_leaf_cert_file <filename>
	}
```

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:84]: This section will need updating as well.

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:89]: This needs updating too

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:98]:
```suggestion
	The default mode is `require_and_verify` if any `trusted_ca_cert` or `trusted_leaf_cert` are provided; otherwise, the default mode is `require`.
```

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:137]: Explain what kind of authentication, or what is necessary and sufficient to auth

File: src/docs/markdown/caddyfile/directives/tls.md
Comment (mholt) [inline src/docs/markdown/caddyfile/directives/tls.md:143]:
```suggestion
		trusted_ca_cert_file   cacerts.crt
```

File: src/docs/markdown/caddyfile/matchers.md
Comment (mholt) [inline src/docs/markdown/caddyfile/matchers.md:126]: This is confusing. Can you not do

```
@exists file {
    try_files ...
}
```
?

**Review by mholt — CHANGES_REQUESTED** [review]
There's a lot of changes here that don't relate to the tls directive. Did you make this? (If not, can you fix that please and reduce this PR to just your changes.)

File: src/docs/markdown/caddyfile/matchers.md
Comment (francislavoie) [reply src/docs/markdown/caddyfile/matchers.md:126]: Yep that's possible in v2.1

**Comment by francislavoie** [comment]
@mholt the reason there's unrelated changes is because we rebased the v2.1 branch and this is still targetting an old copy of the v2.1 branch. This PR needs to be rebased onto the latest version of the v2.1 branch.

**Comment by danlsgiga** [comment]
Do I need to push another PR for the tls client docs?

**Comment by mholt** [comment]
@danlsgiga Yes, please. I don't know why this is closed -- I didn't even open this repo at all today.

Please clean it up and make sure it conforms with https://github.com/caddyserver/caddy/commit/1dfb11486eacc32af1003242023ddc4544823a31 -- thank you!

**Comment by francislavoie** [comment]
I think it was automatically closed because you merged the v2.1 branch actually. Didn't realize that's what GitHub does in those situations but it makes sense I guess.

**Pull request closed without merging**
